File: store/src/app/shared/order.ts
export class Order {
    //check field names
    orderId: number;
    userName: string;
    orderDate: string;
    totalPrice: number;

    constructor(orderId: number, userName: string, orderDate: string, totalPrice: number) {
        this.orderId = orderId;
        this.userName = userName;
        this.orderDate = orderDate;
        this.totalPrice = totalPrice;
    }
}

File: Basket/Basket/Models/Order.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using System.ComponentModel.DataAnnotations;

namespace Basket.Models
{
    public class Order
    {
        [Key]
        public int OrderId { get; set; }
        public string UserName { get; set; }
        public System.DateTime OrderDate { get; set; }
        public decimal TotalPrice { get; set; }
    }
}
